compiler/rustc_typeck/src/check/dropck.rs

```diff
@@ -226,12 +226,14 @@ fn ensure_drop_predicates_are_implied_by_item_defn<'tcx>(
         // could be extended easily also to the other `Predicate`.
         let predicate_matches_closure = |p: Predicate<'tcx>| {
             let mut relator: SimpleEqRelation<'tcx> = SimpleEqRelation::new(tcx, self_param_env);
+            let bound_predicate = predicate.bound_atom(tcx);
+            let bound_p = p.bound_atom(tcx);
             match (predicate.skip_binders(), p.skip_binders()) {
```

Comment:
nit: style

maybe instead shadow `p` and `predicate` here and match on `(predicate.skip_binder(), p.skip_binder())`